src/v/cloud_storage/remote.h

```diff
@@ -21,8 +21,26 @@
 #include <seastar/core/gate.hh>
 #include <seastar/core/loop.hh>
 
+#include <utility>
+
 namespace cloud_storage {
 
+/// \brief Predicate required to continue operation
+///
+/// Describes a predicate to be evaluated before starting an expensive
+/// operation, or to be evaluated when an operation has failed, to find
+/// the reason for failure
+struct should_continue_operation {
+    ss::sstring stop_condition;
+    ss::noncopyable_function<bool()> predicate;
+    should_continue_operation(
+      ss::sstring description, ss::noncopyable_function<bool()> predicate)
+      : stop_condition(std::move(description))
+      , predicate(std::move(predicate)) {}
+
+    bool operator()() const { return predicate(); }
```

Comment:
Let's name it `bool abort_requested()` to be consistent with `abort_source`.

Reply:
In this suggestion, would the semantics of the predicate be reversed, IE the normal course of operation where we can progress is `!abort_requested` and the special case where we have to abort is `abort_requested`?

eg: 

```
auto abort_requested = [this, original_term] {
          return !_partition->is_leader() || _partition->term() != original_term;
      };
```

I think `abort_requested` is a little misleading as no one is actively requesting an abort, it should be something like `must_abort`?

Reply:
makes sense, will update, thanks

Reply:
updated in latest commit